Lib/test/_test_multiprocessing.py

```diff
@@ -3610,6 +3616,246 @@ def test_copy(self):
         self.assertAlmostEqual(bar.y, 5.0)
         self.assertEqual(bar.z, 2 ** 33)
 
+
+class _TestSharedMemory(BaseTestCase):
+
+    ALLOWED_TYPES = ('processes',)
+
+    def setUp(self):
+        if not HAS_SHMEM:
+            self.skipTest("requires multiprocessing.shared_memory")
```

Comment:
You can decorate the class instead

Reply:
Most of the other tests in `_test_multiprocessing.py` use this (potentially older) style rather than the decorator, but this is no longer universally true.  I will adopt the decorator style for new tests added to the testing module.  Changed in commit 1e5341eb171d7d2b988880020cfcc0a64021326d as well.